### ISSUE TITLE
TypeScript .d.ts files from the javascript

### ISSUE THREAD

**Issue by garyhuntddn** [jitsi/jitsi-meet-logger#17]
[garyhuntddn:generate-ts-definitions](https://github.com/jitsi/jitsi-meet-logger/pull/16)

@saghul this is a quick update to create the .d.ts files automatically from the javascript sources

what's the build and publish process for this code? I'm not experienced in publishing these so I may need a bit of hand-holding (most of my work over the last 30 years is internally focussed and we use gitlab rather than github so there's a few things that keep catching me out)

there are some settings in the tsconfig.json which we'll hopefully be able to remove - e.g. the out folder

**Comment by garyhuntddn** [comment]
Just spotted the autobuild github action.

I don't think we want the .d.ts files in github - we'd be better building them each time.

So I believe we just need to add the ```tsc --emitDeclarationOnly``` line just before line 41 of the build script.

How does that sound?

**Comment by garyhuntddn** [comment]
PS I also spotted that the package.json dependencies aren't pinned - I guess if it's your policy, that should be done too

**Comment by saghul** [comment]
> I don't think we want the .d.ts files in github - we'd be better building them each time.

Agreed, we can make that part of the `prepack` phase with an npm script.

**Comment by saghul** [comment]
In the `scripts` section: `"prepack": "npm run create-types",`

**Comment by garyhuntddn** [comment]
That's in place so what are the next steps?

**Comment by saghul** [comment]
Sorry for the delay, I'll try to get to the rewrite PR this week.
